File: src/main/scala/org/zezutom/learningspark/scala/Accum.scala
package org.zezutom.learningspark.scala

import org.apache.spark.Accumulator
import org.apache.spark.rdd.RDD

object Accum {
  def sum(nums: RDD[Int], accum: Accumulator[Int]): Int = {
    nums.foreach(x => accum += x)
    accum.value
  }
}

File: src/main/java/org/zezutom/learningspark/Accum.java
package org.zezutom.learningspark;

import org.apache.spark.Accumulator;
import org.apache.spark.api.java.JavaRDD;

public class Accum {

    public static int sum(JavaRDD<Integer> nums, Accumulator<Integer> accum) {
        nums.foreach(x -> accum.add(x));
        return accum.value();
    }
}
